src/Client/BuzzHouse/Generator/SQLTable.cpp

```diff
@@ -1500,32 +1460,26 @@ const std::vector<TableEngineValues> like_engs
        TableEngineValues::Merge,
        TableEngineValues::Distributed};
 
+static const auto replace_table_lambda = [](const SQLTable & t)
+{ return (!t.db || t.db->attached == DetachStatus::ATTACHED) && t.attached == DetachStatus::ATTACHED && !t.hasDatabasePeer(); };
```

Comment:
I still see 6 references to this same code. IMO it's worth moving the check to its own function and build the rest on top of it, but I'm not going to insist anymore on this if you think it's ok:

```cpp
!t.db || t.db->attached == DetachStatus::ATTACHED) && t.attached == DetachStatus::ATTACHED
```